docs/source/notes/cuda.rst

```diff
@@ -511,6 +511,61 @@ of the alloc/free functions that match the signatures specified above.
    # This will error since the current allocator was already instantiated
    torch.cuda.memory.change_current_allocator(new_alloc)
 
+An alternative approach is to use the :mod:`torch.utils.cpp_extension` to easily define
+allocators directly in the same python source and let PyTorch to take care of the
+compilation process.
+
+.. code:: python
+
+    import torch
+    from torch.utils import cpp_extension
+    
+    custom_allocator_code = """
+    #include <sys/types.h>
+    #include <cuda_runtime_api.h>
+    #include <torch/extension.h>
+    
+    extern "C" {
+    int my_malloc(void** ptr, ssize_t size, int device, cudaStream_t stream) {
+       int err = cudaMalloc(ptr, size);
+       std::cout<<"alloc "<<ptr<<" "<<size<<" "<<err<<std::endl;
+       return err;
+    }
+    
+    int my_free(void* ptr, ssize_t size, int device, cudaStream_t stream) {
+       int err = cudaFree(ptr);
+       std::cout<<"free "<<ptr<< " "<<err<<std::endl;
+       return err;
+    }
+    }
+    
+    // Needed for torch extension to work
+    PYBIND11_MODULE(TORCH_EXTENSION_NAME, m) {}
```

Comment:
You should provide the `functions=` argument to load_inline to ensure they are properly exposed?

Reply:
adding that argument generates python bindings for the functions.
But we want the C++ function address to hook it in the c10 allocator, so exposing them won't give us this pointer :(

Reply:
It will expose them to the python side but the original will also be available on the .so ?